compiler/modules/types/tests/coretest.bal

```diff
@@ -331,3 +331,23 @@ function testIntSubtypeConstraints() {
     test:assertEquals(c13.max, 15);
     test:assertEquals(c13.all, false);
 }
+
+@test:Config{}
+function testComparableNillableList() {
+    [Range, SemType][] members1 = [[{ min: 0, max: 10 }, INT], [{ min: 10, max: int:MAX_VALUE }, STRING]];
```

Comment:
There's a confusion here: max in a Range in **inclusive** (maximum always implies inclusive).